modern/gcc/Dockerfile

```diff
@@ -61,8 +61,6 @@ RUN sudo rm -rf /usr/lib/gcc/x86_64-linux-gnu/* \
     && sudo cp -a /tmp/install/bin/* /usr/local/bin/ \
     && sudo rm -rf /tmp/install \
     && sudo update-alternatives --install /usr/local/bin/cc cc /usr/local/bin/gcc 100 \
-    && sudo update-alternatives --install /usr/local/bin/cpp cpp /usr/local/bin/g++ 100 \
-    && sudo update-alternatives --install /usr/local/bin/c++ c++ /usr/local/bin/g++ 100 \
```

Comment:
if these links are no longer installed, what package does provide them? which compiler do they actually invoke?

Reply:
Good question. They are not link, as they used to be on legacy images. When building gcc from sources, it generates both cpp and c++ as entire binaries. To exemplify, here is the complete list of binaries that we generate:

```
$ docker run --rm -ti conanio/gcc11-ubuntu16.04:1.47.0 ls -lsa /tmp/install/bin
total 43324
    4 drwxr-xr-x 2 root root     4096 Apr 18 15:48 .
    4 drwxr-xr-x 8 root root     4096 Apr 20 07:54 ..
 1200 -rwxr-xr-x 4 root root  1227840 Apr 18 15:48 c++
 1200 -rwxr-xr-x 1 root root  1227840 Apr 18 15:48 cpp
 1200 -rwxr-xr-x 4 root root  1227840 Apr 18 15:48 g++
 1196 -rwxr-xr-x 3 root root  1223744 Apr 18 15:48 gcc
   32 -rwxr-xr-x 2 root root    31288 Apr 18 15:48 gcc-ar
   32 -rwxr-xr-x 2 root root    31288 Apr 18 15:48 gcc-nm
   32 -rwxr-xr-x 2 root root    31288 Apr 18 15:48 gcc-ranlib
 1312 -rwxr-xr-x 1 root root  1340232 Apr 18 15:48 gcov
  552 -rwxr-xr-x 1 root root   561248 Apr 18 15:48 gcov-dump
  572 -rwxr-xr-x 1 root root   581824 Apr 18 15:48 gcov-tool
 1200 -rwxr-xr-x 2 root root  1227840 Apr 18 15:48 gfortran
28700 -rwxr-xr-x 1 root root 29387368 Apr 18 15:48 lto-dump
 1200 -rwxr-xr-x 4 root root  1227840 Apr 18 15:48 x86_64-linux-gnu-c++
 1200 -rwxr-xr-x 4 root root  1227840 Apr 18 15:48 x86_64-linux-gnu-g++
 1196 -rwxr-xr-x 3 root root  1223744 Apr 18 15:48 x86_64-linux-gnu-gcc
 1196 -rwxr-xr-x 3 root root  1223744 Apr 18 15:48 x86_64-linux-gnu-gcc-11.1.0
   32 -rwxr-xr-x 2 root root    31288 Apr 18 15:48 x86_64-linux-gnu-gcc-ar
   32 -rwxr-xr-x 2 root root    31288 Apr 18 15:48 x86_64-linux-gnu-gcc-nm
   32 -rwxr-xr-x 2 root root    31288 Apr 18 15:48 x86_64-linux-gnu-gcc-ranlib
 1200 -rwxr-xr-x 2 root root  1227840 Apr 18 15:48 x86_64-linux-gnu-gfortran
```

Reply:
As you can see, none of them is a link. Also, only `cc` is missing.

Reply:
okay, so there is no `/usr/bin/cc`? just `/usr/local/bin/cc` we create?

Reply:
Correct, it does not exist. But it's not a problem, but we updated PATH to use `/usr/local/bin' first, so any compiler stuff will be collected from there.

Reply:
okay, if it takes a priority, I am fine with it